### PR TITLE
grafana: 8.1.6 -> 8.2.3, fix CVE-2021-41174, enable `buildGo117Module`

### PR DESCRIPTION
**Pull request by Ma27**
###### Motivation for this change

* buildGo117{Package,Module}: enable

    Required to bump Grafana to 8.2.3 in order to fix CVE-2021-41174.
    Building it works fine on both `x86_64-linux` and `aarch64-linux`, so
    the only blocker are incompatible hashes.

    As long as `buildGoModule` doesn't default to Go 1.17 without a
    coordinated update, this should be OK.

* grafana: 8.1.6 -> 8.2.3, fix CVE-2021-41174

  ChangeLogs:
  * https://github.com/grafana/grafana/releases/tag/v8.1.7
  * https://github.com/grafana/grafana/releases/tag/v8.2.0
  * https://github.com/grafana/grafana/releases/tag/v8.2.1
  * https://github.com/grafana/grafana/releases/tag/v8.2.2
  * https://github.com/grafana/grafana/releases/tag/v8.2.3

  Apart from a few new features, most notably is CVE-2021-41174[1]. Darwin
  is temporarily marked as unsupported because Go 1.17 doesn't support the
  Apple SDK versions used in `nixpkgs`.

  [1] https://nvd.nist.gov/vuln/detail/CVE-2021-41174

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-burns** [comment]
So uh, I just uncommented go's `badPlatforms = [ "x86_64-darwin" ]` and it built for me without any problems on x86_64-darwin. Checked out this PR, reenabled grafana on darwin and it builds too. Does go completely do its own thing and not care about our stdenv deployment target, and instead only depend on a minimum host OS version? Maybe the whole go 1.17 issue was a nothingburger.

Anyway it's past my bedtime. Please someone tell me if I'm hallucinating here.

**Comment by Ma27** [comment]
> Anyway it's past my bedtime. Please someone tell me if I'm hallucinating here.

No idea, I just re-read the original discussion and Go's 1.17 changelog also states that a somewhat newer SDK is newer, however I have almost no idea about x86_64-darwin internals, so I'm perhaps missing some context :) 

Anyway, if Go 1.17 works fine for everyone and the Go codeowner agree, I don't have a problem with unmarking x86_64-darwin as broken, of course :)

**Comment by Ma27** [comment]
ping @zowoq @mic92 @kalbasit

**Comment by r-burns** [comment]
Okay, go_1_17 has been unmarked broken on x86_64-darwin so you can do away with the darwin disablement in pkgs/development/go-modules/generic/default.nix and grafana.meta.

**Review by urbas — APPROVED** [review]

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -20254,10 +20254,10 @@ with pkgs;
...
   # https://github.com/NixOS/nixpkgs/issues/144667
   #
-  # do not uncomment this without approval from the go CODEOWNERS
-  #buildGo117Module = callPackage ../development/go-modules/generic {
-  #  go = buildPackages.go_1_17;
-  #};
+  # That's why `buildGoModule != buildGo117Module`.
+  buildGo117Module = callPackage ../development/go-modules/generic {
+    go = buildPackages.go_1_17;
+  };
```
Comment (r-burns) [inline pkgs/top-level/all-packages.nix:20260]: This was how I understood this part, at least. Would appreciate an ACK from a Go codeowner.

**Review by r-burns — APPROVED** [review]
Result of `nixpkgs-review pr 144588` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>grafana</li>
  </ul>
</details>

Result of `nixpkgs-review pr 144588` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>grafana</li>
  </ul>
</details>

**Comment by Ma27** [comment]
Given what we know now, enabling `buildGo117`Module` - without making it the default - should be OK. Also, I stumble upon more and more stuff that actually needs Go 1.17, so I'll merge it in ~48 hours unless there are any objections.

cc @Mic92 @zowoq @kalbasit

**Pull request merged**